torch/fx/experimental/symbolic_shapes.py

```diff
@@ -515,8 +516,8 @@ def guard_scalar(a):
 def _constrain_symbol_range(shape_env, s: sympy.Symbol, compiler_min: int, compiler_max: int):
     upd_vr = ValueRanges(compiler_min, compiler_max)
     old_vr = shape_env.var_to_range.get(s, ValueRanges.unknown())
-    new_vr = shape_env.var_to_range[s] = old_vr & upd_vr
-    if new_vr != old_vr:
+    shape_env._update_var_to_range(s, upd_vr)
```

Comment:
If not to do & upd_vr. May not we loose some constraints information from old_vr?

Reply:
I see - it will be done in _update_var_to_range with previous value